cache_to_disk/__init__.py

```diff
@@ -71,37 +182,51 @@ def delete_old_disk_caches():
     new_cache_metadata = deepcopy(cache_metadata)
     cache_changed = False
     for function_name, function_caches in cache_metadata.items():
-        if function_name != 'total_number_of_cache_to_disks':
-            to_keep = []
-            for function_cache in function_caches:
-                max_age_days = int(function_cache['max_age_days'])
-                file_name = disk_cache_dir + function_cache['file_name']
-                if file_exists(file_name):
-                    if get_age_of_file(file_name) > max_age_days:
-                        cache_changed = True
-                        os.remove(file_name)
-                    else:
-                        to_keep.append(function_cache)
-                else:
-                    cache_changed = True
-            if to_keep:
-                new_cache_metadata[function_name] = to_keep
+        if function_name == _TOTAL_NUMCACHE_KEY:
+            continue
+        to_keep = []
+        for function_cache in function_caches:
+            max_age_days = int(function_cache['max_age_days'])
+            file_name = join_path(DISK_CACHE_DIR, function_cache['file_name'])
+            if not file_exists(file_name):
+                cache_changed = True
+                continue
+            if not get_age_of_file(file_name) > max_age_days != UNLIMITED_CACHE_AGE:
```

Comment:
I think this is a bit more readable: 

if (get_age_of_file(file_name) <= max_age_days) or (max_age_days == UNLIMITED_CACHE_AGE)

Reply:
> This is brilliant, thank for modernizing this! I'm happy to keep your doc string in there and all of the changes look good. Have you been using this code and tested it?

I have been using it for only a few days, in a single project. I haven't noticed any issues so far, but as you can tell I didn't add any test-suites. I'm just monitoring the `cache_info()` while my app runs, to make sure it looks good, and occasionally using `cache_clear()` when I need to test new logic in the cached functions. So far it's worked great